src/otx/algo/detection/yolov9.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""YOLOv9 model implementations."""
+
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING, Literal
+
+from otx.algo.detection.backbones import GELAN
+from otx.algo.detection.detectors import SingleStageDetector
+from otx.algo.detection.heads import YOLOHead
+from otx.algo.detection.losses.yolov9_loss import BCELoss, BoxLoss, DFLoss, YOLOv9Criterion
+from otx.algo.detection.necks import YOLONeck
+from otx.algo.detection.utils.yolov7_v9_utils import Vec2Box
+from otx.core.config.data import TileConfig
+from otx.core.exporter.base import OTXModelExporter
+from otx.core.exporter.native import OTXNativeModelExporter
+from otx.core.metrics.fmeasure import MeanAveragePrecisionFMeasureCallable
+from otx.core.model.base import DefaultOptimizerCallable, DefaultSchedulerCallable
+from otx.core.model.detection import OTXDetectionModel
+
+if TYPE_CHECKING:
+    from lightning.pytorch.cli import LRSchedulerCallable, OptimizerCallable
+    from typing_extensions import Self
+
+    from otx.core.metrics import MetricCallable
+    from otx.core.schedulers import LRSchedulerListCallable
+    from otx.core.types.label import LabelInfoTypes
+
+
+PRETRAINED_WEIGHTS: dict[str, str] = {
```

Comment:
We have difference here across our tasks. I personally, add this as a class attribute, but you add it as a global parameter. We need to have common decision

Reply:
I agree, maybe it seems because we did refactoring parallel.
I referred vit for classification.
https://github.com/openvinotoolkit/training_extensions/blob/709d841f4a2c9e8dc73b5e416a1deb7536a6b761/src/otx/algo/classification/vit.py#L48-L63

I'll unify this pretrained weight list location for all tasks in the next PR.